vulkan_helpers/helper_functions.cpp

```diff
@@ -1142,10 +1140,12 @@ VkSwapchainKHR CreateDefaultSwapchain(
     }
 
     uint32_t minSwapchains =
-        std::max(surface_caps.minImageCount + 1, min_image_count + 1);
+        min_image_count ? min_image_count : surface_caps.minImageCount + 1;
+    minSwapchains = std::max(min_image_count, surface_caps.minImageCount);
 
-    uint32_t numSwapchain =
-        std::min(surface_caps.maxImageCount, minSwapchains);
+    uint32_t numSwapchain = std::min(surface_caps.maxImageCount, minSwapchains);
```

Comment:
It looks like surface_caps.maxImageCount can be 0, meaning there is no limit: 
"A value of 0 means that there is no limit on the number of images, though there may be limits related to the total amount of memory used by presentable images."
https://www.khronos.org/registry/vulkan/specs/1.2-extensions/man/html/VkSurfaceCapabilitiesKHR.html
Do we need to cover that case?

Reply:
Thanks! Fixed now.